File: samples/python/07-Simple-Native-Functions/plugins/MathPlugin/Math.py
import math
from semantic_kernel.skill_definition import sk_function


class Math:
    @sk_function(
        description="Takes the square root of a number",
        name="Sqrt",
        input_description="The value to take the square root of",
    )
    def square_root(self, number: str) -> str:
        return str(math.sqrt(float(number)))

File: samples/dotnet/10-Chaining-Functions/plugins/MathPlugin/Math.cs
// Copyright (c) Microsoft. All rights reserved.

using System.ComponentModel;
using System.Globalization;
using Microsoft.SemanticKernel;

namespace Plugins.MathPlugin;

public class Math
{
    [SKFunction, Description("Take the square root of a number")]
    public static string Sqrt(string number)
    {
        return System.Math.Sqrt(Convert.ToDouble(number, CultureInfo.InvariantCulture)).ToString(CultureInfo.InvariantCulture);
    }

    [SKFunction, Description("Multiply two numbers")]
    public static double Multiply(
        [Description("The first number to multiply")] double input,
        [Description("The second number to multiply")] double number2
    )
    {
        return input * number2;
    }
}
